src/main/java/seedu/duke/Forex.java

```diff
@@ -17,35 +22,32 @@ public class Forex {
      * convert first to SGD, then from SGD to their target. To be
      * implemented using API.
      */
-    private static final HashMap<Currency, BigDecimal> exchangeRates = new HashMap<Currency, BigDecimal>();
+    private static HashMap<Currency, BigDecimal> exchangeRates;
 
     private Currency initial;
     private Currency target;
 
     // Constructor
     public Forex (Currency from, Currency to) {
-        populateRates();
         initial = from;
         target = to;
     }
 
     /**
-     * Populates the exchange rates hash map with the current exchange rates,
-     * correct as of 2023-03-15 18:15 SGT according to
-     * <a href="https://www.xe.com/currencyconverter/convert">https://www.xe.com/currencyconverter/convert</a>
-     */
-    public static void populateRates () {
-        exchangeRates.put(Currency.SGD, BigDecimal.valueOf(1.000f));
-        exchangeRates.put(Currency.MYR, BigDecimal.valueOf(3.3295727f));
-        exchangeRates.put(Currency.USD, BigDecimal.valueOf(0.74265183f));
-        exchangeRates.put(Currency.IDR, BigDecimal.valueOf(11442.776f));
-        exchangeRates.put(Currency.JPY, BigDecimal.valueOf(99.414735f));
-        exchangeRates.put(Currency.THB, BigDecimal.valueOf(25.689173f));
-        exchangeRates.put(Currency.CNY, BigDecimal.valueOf(5.1222378f));
-        exchangeRates.put(Currency.GBP, BigDecimal.valueOf(0.61293137f));
-        exchangeRates.put(Currency.EUR, BigDecimal.valueOf(0.69567172f));
-        exchangeRates.put(Currency.KRW, BigDecimal.valueOf(974.90478f));
-        exchangeRates.put(Currency.VND, BigDecimal.valueOf(17490.625f));
+     * Initializes the exchange rates by fetching them from the ExchangeRates class.
+     * Buffers for 5 seconds to allow the API call to go through.
+     * This method will only be called once upon starting the program.
+     */ 
+    public static void initializeRates() {
+        Ui ui = new Ui();
+        ExchangeRates initRates = new ExchangeRates();
```

Comment:
I think we can remove this variable if it is not used

Reply:
This variable cannot be removed. The constructor for ExchangeRates calls fetchExchangeRates, which makes the API calls. Creating this object is equivalent to calling ExchangeRates.fetchExchangeRates(), but I want to keep access to fetchExchangeRates() private.